### DISCUSSION TITLE
Scrump vs Stump

### DISCUSSION THREAD

**Discussion by OriginalGoku** [TDAmeritrade/stumpy#863] [General]
Hello everyone,
I have recently found out about Stumpy and I can't believe its usefulness in time series analysis. I have developed many applications using deep learning and reinforcement learning over the years but this simple tool seems to be an amazing companion (even better overall) if you factor in the costs of research and development of such complex systems. 
Also, I wanted to thank Sean and the rest of the team for writing such detailed tutorials. I must have read them at least 5 times so far to fully understand every aspect of the code. 

Now my question is regarding the performance increase by using stump vs scrump. I am using a dataset of roughly 6000 rows for my data. 

```
scrump_time = []
stump_time = []
for i in range(100):
  # Time the stump method
  start_time = time.time()
  mp = stumpy.stump(data, m)
  true_P = mp[:, 0]
  stump_time.append(time.time() - start_time)

  # Time the scrump method
  start_time = time.time()
  approx = stumpy.scrump(data, m, percentage=0.1, pre_scrump=True, s=None)
  approx.update()
  approx_P = approx.P_
  scrump_time.append(time.time() - start_time)

# Print the timing results
print("Scrump time :", np.mean(scrump_time))
print("Stump time :", np.mean(stump_time))
```
and here are the results:
```
Scrump time (approximate): 3.168635606765747
Stump time (Original): 0.5034401416778564
```
I am using Colab to test this but it was roughly the same on a computer without GPU. 
I have read the scrump test case file but that's all with randomly generated data so I decided to test this on my own time series data (the date is basically the closing price of SPY).
Can you please explain how to really make the best use of scrump function?

**Comment by seanlaw** [comment]
@OriginalGoku Thank you for your kind words. Realistically, `scrump` could/should be used when you have much longer time series (i.e., 1,000,000+ i length) and it may not be feasible to compute the full matrix profile and an approximate ("good enough") matrix profile is sufficient for your needs. As a general rule of thumb, if the time series is less than 10,000 data points, it probably makes sense to use `stump` because it will be "fast enough" for the majority of cases. If you need the full (accurate) matrix profile then you'll want to stick to `stump` instead of `scrump`. Let me know if that makes sense.

**Reply by OriginalGoku** [reply]
Hi Sean,
That makes perfect sense. Thanks for your prompt reply. 
I am part of a team of AI engineers who have been working on time series for years and we can really appreciate this efficient and well-documented tool.
I am very busy with my work but I will try to contribute to this project as soon as I have some extra time as a small token of appreciation. 
Once again, thank you and the rest of the team for such beautiful work.

**Reply by seanlaw** [reply]
> I will try to contribute to this project as soon as I have some extra time as a small token of appreciation.

I'm sure the community would welcome that! We're always looking for new and dedicated contributors

> Once again, thank you and the rest of the team for such beautiful work.

Please let us know if there is any additional feedback/criticism. We're always looking for ways to better serve our community!
